compiler/noirc_frontend/src/tests.rs

```diff
@@ -3389,3 +3389,14 @@ fn arithmetic_generics_rounding_fail() {
     let errors = get_program_errors(src);
     assert_eq!(errors.len(), 1);
 }
+
+#[test]
+fn unconditional_recursion() {
```

Comment:
The tests here (in frontend/tests) don't run the full compiler, just the frontend. So all the SSA code doesn't run here. That's why this doesn't error... but then maybe this test shouldn't be here (maybe it should be in `test_programs` in `compile_failure)`.

Reply:
Thanks, I added this because I didn't know about the SSA at all. Then I added tests where they are executed, and figured out where to do the other test above in `inlining.rs`. I thought I would leave this here as a demonstration that this thing is _not_ caught at the moment.

However in https://github.com/noir-lang/noir/pull/6292 I replaced it with failure and successful scenarios after implementing this error detection in the frontend, so we can catch the obvious case early. 

I don't have an opinion on whether it should live here or in the `compile_failure` directory 🤷 

FWIW I collected these places where I look for these kinds of tests, I thought it would be a good addition to the README:

```
Tests:
- tooling/nargo_cli/build.rs                         <- run tests in test_programs
- tooling/nargo_cli/tests/stdlib-builtin.rs          <- run tests in std
- tooling/nargo_cli/tests/stdlib-props.rs            <- fuzz testing code snippets
- tooling/nargo_fmt/tests/execute.rs                 <- expected formatter outcomes
- compiler/noirc_frontend/src/hir/comptime/tests.rs  <- interpreted program snippets
- compiler/noirc_frontend/src/lexer/lexer.rs         <- lexer unit tests
- compiler/noirc_frontend/src/parser/parser/*.rs     <- parser unit tests
- compiler/noirc_frontend/src/tests.rs               <- compilation
- compiler/noirc_frontend/src/tests/*                <- compilation by feature
```

Reply:
I think leaving it here might be misleading (when quickly reading it it might seem that infinite recursion does not lead to an error).